tests/components/deconz/test_fan.py

```diff
@@ -0,0 +1,185 @@
+
+    # Service turn off fan
+
+    with patch.object(
+        ceiling_fan_device, "_request", return_value=True
+    ) as set_callback:
+        await hass.services.async_call(
+            fan.DOMAIN,
+            fan.SERVICE_TURN_OFF,
+            {"entity_id": "fan.ceiling_fan"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        set_callback.assert_called_with("put", "/lights/1/state", json={"speed": 0})
+
+    # Service set fan speed to low
+
+    with patch.object(
+        ceiling_fan_device, "_request", return_value=True
+    ) as set_callback:
+        await hass.services.async_call(
+            fan.DOMAIN,
+            fan.SERVICE_SET_SPEED,
+            {"entity_id": "fan.ceiling_fan", fan.ATTR_SPEED: fan.SPEED_LOW},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        set_callback.assert_called_with("put", "/lights/1/state", json={"speed": 1})
+
+    # Service set fan speed to medium
+
+    with patch.object(
+        ceiling_fan_device, "_request", return_value=True
+    ) as set_callback:
+        await hass.services.async_call(
+            fan.DOMAIN,
+            fan.SERVICE_SET_SPEED,
+            {"entity_id": "fan.ceiling_fan", fan.ATTR_SPEED: fan.SPEED_MEDIUM},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        set_callback.assert_called_with("put", "/lights/1/state", json={"speed": 2})
+
+    # Service set fan speed to high
+
+    with patch.object(
+        ceiling_fan_device, "_request", return_value=True
+    ) as set_callback:
+        await hass.services.async_call(
+            fan.DOMAIN,
+            fan.SERVICE_SET_SPEED,
+            {"entity_id": "fan.ceiling_fan", fan.ATTR_SPEED: fan.SPEED_HIGH},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        set_callback.assert_called_with("put", "/lights/1/state", json={"speed": 4})
+
+    # Service set fan speed to off
+
+    with patch.object(
+        ceiling_fan_device, "_request", return_value=True
+    ) as set_callback:
+        await hass.services.async_call(
+            fan.DOMAIN,
+            fan.SERVICE_SET_SPEED,
+            {"entity_id": "fan.ceiling_fan", fan.ATTR_SPEED: fan.SPEED_OFF},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        set_callback.assert_called_with("put", "/lights/1/state", json={"speed": 0})
+
+    # Verify that an unsupported speed gets converted to default speed "medium"
+
+    state_changed_event = {
+        "t": "event",
+        "e": "changed",
+        "r": "lights",
+        "id": "1",
+        "state": {"speed": 3},
+    }
+    gateway.api.event_handler(state_changed_event)
+    await hass.async_block_till_done()
+
+    assert hass.states.get("fan.ceiling_fan").state == STATE_ON
+    assert hass.states.get("fan.ceiling_fan").attributes["speed"] == fan.SPEED_MEDIUM
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)